### ISSUE TITLE
NVDA can't catch aria-live attribute with chrome in certain situation with button text

### ISSUE THREAD

**Issue by khsbory** [nvaccess/nvda#11798]
Hello, my name is Hyongsop Kim from Korea and NVDA user.
In [this sample page](https://khsbory.github.io/video%20player%20with%20aria-live%20script.html), there is a play button in the main contents.
Once I press the button, the button text is changed to pause.
The button text is just text, not value or aria-label, NVDA read nothing once the button text is changed.
So I added aria-live with polite attribute once button text is changed.
By the way, after page loads and I press the play button, NVDA can't catch aria-live attribute and read nothing.
From second press, NVDA read the changed text and catch aria-live.
As a result, once DOM loads or refreshed, NVDA does not recognize aria-live attribute first and from the second, it recognize aria-live.
You can watch this video for your reference.
I recorded [this video](https://1drv.ms/v/s!Aj9xs5XCa0zwmK5isvb2BcuioRGq6Q?e=i3Om2C) for demonstration purposes.
Please check this issue.
Thank you.

**Comment by Adriani90** [comment]
cc: @feerrenrut, @derekriemer I recall there was  a PR to improve Aria live pollite, refactoring the NVDA Live helper. But I cannot find it.

**Comment by feerrenrut** [comment]
I'm not sure that aria-live is the right choice in this case, but assuming this is what you want to do, there are some issues with the sample, be sure to consult [MDN for aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions).

For aria-live to work the content that will be changed needs to be within an aria-live element, and the value of aria-live should not be `true`, but instead one of the values  `off, polite or assertive`

Here is a working (tested with firefox 82.0.2 and chrome 86.0.4240.183) [sample](https://codepen.io/reefturner/full/mdEzOPz) and the [code pen source](https://codepen.io/reefturner/pen/mdEzOPz?editors=1010) for it.

I'm going to assume we can close this issue, please comment if there are still issues.

**Comment by khsbory** [comment]
Hello.
Thank you for your reply and testing with sample page.
In case of your sample, you wrote aria-live attribute inside div with button.
However in that case there is a problem.
In case of iOS VoiceOver, if I go to your sample with iOS, VoiceOver will read hello button as soon as page loads even if it is not VoiceOver focused to button.
Because prior to press the hello button, aria-live is already added DOM and VoiceOver should read this.
But the reason to add aria-live attribute with this button is that once user press the button, let screen reader read changed text.
In other words, in your source, like VoiceOver, NVDA should read the hello button as soon as the page load, because div has aria-live attribute as soon as page load.
But NVDA doesn't read the hello button and once I press the button, NVDA only then read the goodbye text.
How do you think about that?

**Comment by feerrenrut** [comment]
I think the behavior of voiceOver is wrong. But as I mentioned, is aria-live really what you want for this case? If not, there are a few other ways to do this, the closest to your initial example is this:

Note this uses `innerText` rather than `innerHTML`. Given you are not actually adding a HTML attribute, this is more appropriate. I have updated the [codepen ](https://codepen.io/reefturner/pen/mdEzOPz?editors=1010) with several suggestions that you can [play with here](https://codepen.io/reefturner/full/mdEzOPz).
``` html
<div>
  <button id="myButton3" onclick="buttonPressed3();">
    hello3
  </button>
<div>
```
``` javascript
function buttonPressed3() {
  b = document.getElementById("myButton3");
  if (b.innerText == "hello3") {
    b.innerText = "good bye";
  } else {
    b.innerText = "hello3";
  }
};
```

**Comment by khsbory** [comment]
Thank you for your sample code and testing.
Do you think it is right for the screen reader to speak only when the content is updated even if the page loads with the aria-live property added to div with button text from the beginning?

**Comment by feerrenrut** [comment]
Ah yes, it looks like I had some errors there, and you are right the issue is no longer fixed.  This will require more investigation. 

>Do you think it is right for the screen reader to speak only when the content is updated even if the page loads with the aria-live property added to div with button text from the beginning?

Yes, the [spec ](https://www.w3.org/TR/wai-aria-1.1/#aria-live) talks about updates, suggesting that the initial content should not be announced.

**Comment by feerrenrut** [comment]
Also discussed here: https://github.com/nvaccess/nvda/issues/7906#issuecomment-807957477
